messaging/FCMSwift/AppDelegate.swift

```diff
@@ -45,9 +45,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
       let settings: UIUserNotificationSettings =
       UIUserNotificationSettings(forTypes: [.Alert, .Badge, .Sound], categories: nil)
       application.registerUserNotificationSettings(settings)
-      application.registerForRemoteNotifications()
     }
 
+    application.registerForRemoteNotifications()
```

Comment:
Shouldn't this be in the `requestAuthorizationWithOptions`'s `completionHandler` instead? Moreso checking it's `granted` parameter also?